src/angular-app/languageforge/lexicon/editor/editor.component.ts

```diff
@@ -128,6 +148,9 @@ export class LexiconEditorController implements angular.IController {
       if (this.hasUnsavedChanges()) {
         this.saveCurrentEntry();
       }
+      // destroy listeners when leaving editor page
+      angular.element(window).unbind('keyup', (e: Event) => {});
+      this.$scope.$destroy();
```

Comment:
Unfortunately `$window.onbeforeunload()` is what is called when a tab or windows is closed.  So this code doesn't do anything helpful before the event handler will be destroyed as the window/tab is closed anyway.

`this.$scope.$destroy();` seems unnecessary to me.  AngularJS should be calling this automatically and we shouldn't need to explicitly call it.

Keeping with the $onInit() lifecycle hook, we should use the complementary `$onDestroy()` hook to clean up the event listener.  So the `element.unbind()` should be done inside the `$onDestoy()`.

Reply:
I did look at $onDestroy() and it does not get called on just leaving and navigating to another page. I thought we discussed not having the key handler listening after leaving the editor page. this.$scope.$destroy(); was just an assurance everything was cleaned up. I can remove it.

Reply:
@palmtreefrb I thought Chris' idea about the locationChange was pretty good, did you find it not to be a good option?

Reply:
Not a good option because it doesn't get called on page load. You have to click on a navigation button to execute the event.

Reply:
ok, yeah that makes sense.  Thank you.

Reply:
that's too bad that `$locationChangeStart` doesn't get fired on first page load :(

Reply:
> since you've requested Chris to approve and the last thing he did was request changes, you either need to remove him from the reviewers or make the changes he requested (if they're still relevant)

Not an option I can find to clear this up to complete the merge. Maybe my user level does not have these options.